TCMSApp/src/client/app/landing/landing.controller.js

```diff
@@ -0,0 +1,61 @@
+(function () {
+    'use strict';
+
+    angular
+        .module('app.landing')
+        .controller('LandingController', LandingController);
+
+    LandingController.$inject = ['$q', 'dataservice', 'logger', '$scope', '$state', '$rootScope'];
+    /* @ngInject */
+
+    function LandingController($q, dataservice, logger, $scope, $state, $rootScope) {
+        var vm = this;
+        vm.news = {
+            title: 'TCMSApp',
+            description: 'Hot Towel Angular is a SPA template for Angular developers.'
+        };
+        vm.messageCount = 0;
+        vm.people = [];
+        vm.title = 'Landing';
+
+        activate();
+
+        function activate() {
+            var promises = [getMessageCount(), getPeople()];
+
+            return $q.all(promises).then(function () {
+                logger.info('Activated Landing View');
+            });
+        }
+
+        function getMessageCount() {
+            return dataservice.getMessageCount().then(function (data) {
+                vm.messageCount = data;
+                return vm.messageCount;
```

Comment:
hm. don't sure that this return is useful. can you prove the opposite? :)

my proposition is  to get rid of getMessageCount() and getPeople(),

add to promises array [dataservice.getMessageCount(), dataservice.getPeople()] and do vm.messageCount and vm.people in activate function.